sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/PropagateEmptyRelation.scala

```diff
@@ -127,7 +131,13 @@ abstract class PropagateEmptyRelationBase extends Rule[LogicalPlan] with CastSup
       }
 
     // the only case can be matched here is that LogicalQueryStage is empty
-    case p: LeafNode if !p.isInstanceOf[LocalRelation] && isEmpty(p) => empty(p)
+    case p: LeafNode if !p.isInstanceOf[LocalRelation] && isEmpty(p) =>
+      if (p.getTagValue(ROOT_REPARTITION).isEmpty) {
```

Comment:
how can a leaf node be root repartition?

Reply:
it is `LogicalQueryStage` in AQE since the reparition is planned to shuffle